src/main/java/net/neoforged/neoforge/common/extensions/ValueInputExtension.java

```diff
@@ -25,4 +26,14 @@ default Set<String> keySet() {
         //noinspection deprecation
         return self().read(MapCodec.assumeMapUnsafe(CompoundTag.CODEC)).orElseThrow().keySet();
     }
+
+    /**
+     * Read the given {@code child} object from the given {@code key}.
+     *
+     * @param key   the key to read the child from
+     * @param child the child to read from given key
+     */
+    default void readChild(String key, ValueIOSerializable child) {
+        child.deserialize(self().childOrEmpty(key));
```

Comment:
I don't think I would want the child to deserialize an entry if it's empty. Rather, only if the child exists should the deserialization happen, or maybe those should be two separate methods.